src/lib/rnp.c

```diff
@@ -1072,24 +1072,23 @@ rnp_encrypt_file(rnp_ctx_t *ctx, const char *userid, const char *f, const char *
 {
     const pgp_key_t *key;
     const char *     suffix;
-    pgp_io_t *       io;
     char             outname[MAXPATHLEN];
 
-    io = ctx->rnp->io;
     if (f == NULL) {
-        (void) fprintf(io->errs, "rnp_encrypt_file: no filename specified\n");
+        (void) fprintf(ctx->rnp->io->errs, "rnp_encrypt_file: no filename specified\n");
         return RNP_FAIL;
     }
-    suffix = (ctx->armour) ? ".asc" : ".gpg";
     /* get key with which to sign */
     if ((key = resolve_userid(ctx->rnp, ctx->rnp->pubring, userid)) == NULL) {
         return RNP_FAIL;
     }
+    /* generate output file name if needed */
     if (out == NULL) {
+        suffix = (ctx->armour) ? ".asc" : ".gpg";
```

Comment:
I think we need a better method than hard-coding file extensions? I know it's a moved line but maybe for the future.